### ISSUE TITLE
Document Usage of Non-String Nodes

### ISSUE THREAD

**Issue by peterstory** [c0fec0de/anytree#13]
Currently, the documentation only shows initializing `Node` using a `string`. However, it isn't clear whether strings are required. For example, I'm interested in creating a tree out of enumerated type values. But I get a variety of strange behavior when trying to do that.

**Comment by c0fec0de** [comment]
Hi,

I will have a Look.

Can you send me your code?

Cheers

**Comment by peterstory** [comment]
Sure! Here's a simplified snippet showing what I'm trying to do:

```python
from enum import IntEnum

from anytree import Node, RenderTree, Resolver

class Animals(IntEnum):
    Mammal = 1
    Cat = 2
    Dog = 3

ANIMAL_TAXONOMY = Node("ANIMAL")
_MAMMAL = Node(Animals.Mammal, parent=ANIMAL_TAXONOMY)
Node(Animals.Cat, parent=_MAMMAL)
Node(Animals.Dog, parent=_MAMMAL)

r = Resolver()
r.glob(ANIMAL_TAXONOMY, "/ANIMAL/*")
```

Gives me this error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.5/dist-packages/anytree/resolver.py", line 147, in glob
    return self.__glob(node, parts)
  File "/usr/local/lib/python3.5/dist-packages/anytree/resolver.py", line 175, in __glob
    matches = self.__find(node, name, remainder)
  File "/usr/local/lib/python3.5/dist-packages/anytree/resolver.py", line 186, in __find
    if Resolver.__match(name, pat):
  File "/usr/local/lib/python3.5/dist-packages/anytree/resolver.py", line 210, in __match
    return re_pat.match(name) is not None
TypeError: expected string or bytes-like object
```

But if I switch to using `Animals.Mammal.name`, etc., then things work.

**Comment by c0fec0de** [comment]
Fixed in 2.8.0
